zetaclient/context/app.go

```diff
@@ -306,25 +292,33 @@ func (a *AppContext) Update(
 	a.additionalChain = additionalChains
 	a.blockHeaderEnabledChains = blockHeaderEnabledChains
 
+	// update core params for evm chains we have configs in file
+	freshEvmChainParams := make(map[int64]*observertypes.ChainParams)
+	for _, cp := range evmChainParams {
+		_, found := a.config.EVMChainConfigs[cp.ChainId]
+		if !found {
+			a.logger.Warn().
+				Int64("chain.id", cp.ChainId).
+				Msg("Encountered EVM ChainParams that are not present in the config file")
+
+			continue
+		}
+
+		freshEvmChainParams[cp.ChainId] = cp
+	}
+
+	a.evmChainParams = freshEvmChainParams
+
 	// update chain params for bitcoin if it has config in file
-	if a.bitcoinChainParams != nil && btcChainParams != nil {
+	if btcChainParams != nil {
```

Comment:
do we shut down Bitcoin observer/signer if newly fetched `btcChainParams` is nil? or a future plan?

Reply:
Let's tackle this in the following PRs.